source/state_representation/include/state_representation/parameters/Parameter.hpp

```diff
@@ -120,11 +125,25 @@ inline void Parameter<T>::set_value(const T& value) {
   this->value_ = value;
 }
 
+/**
+ * @brief Create a Parameter object that is owned by a shared_ptr
+ * @tparam T The type of the value stored in the parameter
+ * @param name The name of the parameter
+ * @param param_value The value of the parameter
+ * @return A shared_ptr that owns the newly created Parameter object
+ */
 template<typename T>
 static std::shared_ptr<Parameter<T>> make_shared_parameter(const std::string& name, const T& param_value) {
   return std::make_shared<Parameter<T>>(name, param_value);
 }
 
+/**
```

Comment:
I have two TODOs in the brief and return tags here. @eeberhard could you help me with a precise wording here?
I would like to make it clear that this is not just a parameter interface but an parameter interface that holds an actual parameter. Otherwise I find the name of the method to be a bit misleading...